jwst/tweakreg/astrometric_utils.py

```diff
@@ -0,0 +1,184 @@
+import os
+import requests
+
+from astropy import table
+from astropy.table import Table
+from astropy.coordinates import SkyCoord
+from astropy import units as u
+
+from ..resample import resample_utils
+from ..assign_wcs import util as wcsutil
+
+ASTROMETRIC_CAT_ENVVAR = "ASTROMETRIC_CATALOG_URL"
+DEF_CAT_URL = 'http://gsss.stsci.edu/webservices'
+
+if ASTROMETRIC_CAT_ENVVAR in os.environ:
+    SERVICELOCATION = os.environ[ASTROMETRIC_CAT_ENVVAR]
+else:
+    SERVICELOCATION = DEF_CAT_URL
+
+"""
+
+Primary function for creating an astrometric reference catalog.
+
+"""
+
+
+def create_astrometric_catalog(input_models, catalog="GAIADR2", output="ref_cat.ecsv",
+                               gaia_only=False, table_format="ascii.ecsv",
+                               existing_wcs=None, num_sources=None):
+    """Create an astrometric catalog that covers the inputs' field-of-view.
+
+    Parameters
+    ----------
+    input : str, list
```

Comment:
The actual parameter is called `input_models`, not `input` and I believe is supposed to be a list of datamodels, not file names.